### PR TITLE
'value' attribute in [experimental_filter_ability(_active)] can have 'default' value

### PR DESCRIPTION
**Pull request by newfrenchy83**
for answer to an suggestion of @soliton- who think in filter abilitie, value must could check 'default' value of an abilitie/specials directly without have specify this value in filter

### CODE REVIEW

File: data/schema/filters/abilities.cfg [hunk 1]
```diff
@@ -10,7 +10,7 @@
 	{SIMPLE_KEY replacement_type string_list}
 	{SIMPLE_KEY alternative_type string_list}
 	{SIMPLE_KEY active_on ability_context}
-	{SIMPLE_KEY value s_int_range_list}
+	{SIMPLE_KEY value string_list}
```
Comment (soliton-) [inline data/schema/filters/abilities.cfg:13]: Would be nicer to add a type that accepts s_int_range_list and just "default".
Perhaps
```
[type]
    name=s_int_range_list_default
    [union]
        [type]
            link=s_int_range_list
        [/type]
        [type]
            value=default
        [/type]
    [/union]
[/type]
```
works.

**Review by soliton- — CHANGES_REQUESTED** [review]

File: src/utils/config_filters.cpp [hunk 2]
```diff
@@ -121,3 +121,16 @@ bool utils::config_filters::bool_or_empty(const config& filter, const config& cf
 	}
 	return is_matches;
 }
+
+//if filter["value"]="default" then this function check if abilitie using default value or if cfg["value] equals todefault value.
+//if abilitie don't have default value then return false because none default value exist for match
+bool utils::config_filters::default_value(const config & filter, const config & cfg, std::optional<int> def)
```
[the remark this answers is not in the document]
Comment (newfrenchy83) [reply src/utils/config_filters.cpp]: modification dones

File: changelog_entries/filter_value_check_default.md [hunk 3]
```diff
@@ -0,0 +1,2 @@
+### WML Engine
+   * filter 'value' can using "default" value for check if abilities using value by default(explicit 'value' or not)
```
[the remark this answers is not in the document]
Comment (newfrenchy83) [reply changelog_entries/filter_value_check_default.md]: changed

File: data/schema/filters/abilities.cfg (hunk 1 above)
Comment (newfrenchy83) [reply data/schema/filters/abilities.cfg]: done

File: data/schema/filters/abilities.cfg [hunk 4]
```diff
@@ -10,7 +10,7 @@
 	{SIMPLE_KEY replacement_type string_list}
 	{SIMPLE_KEY alternative_type string_list}
 	{SIMPLE_KEY active_on ability_context}
-	{SIMPLE_KEY value s_int_range_list}
+	{SIMPLE_KEY value s_int_range_list_default}
```
Comment (soliton-) [inline data/schema/filters/abilities.cfg:13]: Now `add`, `sub`, `multiply` and `divide` also support "default".

**Review by soliton- — APPROVED** [review]

File: data/schema/filters/abilities.cfg (hunk 4 above)
Comment (newfrenchy83) [reply data/schema/filters/abilities.cfg:13]: No because none had def value specified

Comment (newfrenchy83) [reply data/schema/filters/abilities.cfg:13]: Default chexked with "default" if and only if default value is specified like in "value" attrbute for smoe ablities type but not for add sub etc, add don't support "default"

Comment (newfrenchy83) [reply data/schema/filters/abilities.cfg:13]: added to shema

**Review by soliton- — APPROVED** [review]

**Comment by newfrenchy83** [comment]
@stevecotton could you merge this PR approved by @soliton- , please?

**Pull request merged**